### ISSUE TITLE
Warning instructs using `--extra-checks`, but no equivalent config option exists

### ISSUE THREAD

**Issue by Feuermurmel** [python/mypy#16189]
**Documentation**

`mypy` prints this warning:

```
Warning: --strict-concatenate is deprecated; use --extra-checks instead
```

I'm not passing `--strict-concatenate` on the command line, but I instead have a `setup.cfg` with `strict_concatenate = True`. So naturally, I would expect that the `strict_concatenate` config has been replaced with an `extra_checks` config, but the docs page [The mypy configuration file](https://mypy.readthedocs.io/en/stable/config_file.html) does not mention such an option but also does not mention that `strict_concatenate` has been deprecated.

So, either the docs of config options needs to be updated, or the warning is wrong and should only be triggered when `--strict-concatenate` is passed on the command line.

**Comment by NikolasWolke** [comment]
`--strict-concatenate` is also still mentioned in [the section about stricter options](https://mypy.readthedocs.io/en/stable/existing_code.html#introduce-stricter-options).

**Comment by jahschwa** [comment]
I've only done a quick skim, but it seems like it is indeed implemented; the config parsing logic pulls from the `Options` class which has it:
https://github.com/python/mypy/blob/06a566b81993a711470aa9e80b9d86d1e666b029/mypy/options.py#L219

So this is probably just a documentation thing. I haven't actually tried to test this by crafting a snippet that would trip something under `extra_checks` but my assumption is just adding `extra_checks = true` to your config file should work as expected.

**Comment by glatterf42** [comment]
Adding `extra_checks = true` to my `pyproject.toml` file does indeed work, as in: it removes the deprecation warning and returns the same errors as before with `strict_concatenate = true`. However, my code might coincidentally comply to the checks performed by `extra_checks` in addition to `strict_concatenate`. So it would still be nice to have more fine-grained access or to adapt the [documented list of `strict` options](https://mypy.readthedocs.io/en/stable/existing_code.html#introduce-stricter-options) so that one can turn them on without deprecation warnings :)

**Comment by Feuermurmel** [comment]
Cool! Thanks for improving the documentation! 🧡
